src/modules/UI/scenes/Plugins/EdgeProvider.js

```diff
@@ -493,6 +503,23 @@ export class EdgeProvider extends Bridgeable {
     return orderData
   }
 
+  // Convert an array of currency codes to an array of EdgeTokenIdExtended objects by creating all possible plugin/token combinations, ignoring any plugin passed as filterPlugins.
+  convertCurrencyCodesToEdgeTokenIds(currencyCodes: string[], filterPlugins?: string[] = []): EdgeTokenIdExtended[] {
+    const tokenIds: EdgeTokenIdExtended[] = []
+    for (const pluginId of Object.keys(this._state.core.account.currencyConfig).filter(plugin => !filterPlugins.includes(plugin))) {
+      const { currencyCode, metaTokens } = this._state.core.account.currencyConfig[pluginId].currencyInfo
+      if (currencyCodes.includes(currencyCode)) {
+        tokenIds.push({ pluginId, currencyCode })
+      }
+      for (const allowedCode of currencyCodes) {
+        if (metaTokens.find(token => token.currencyCode === allowedCode) != null) {
```

Comment:
This will not work for custom tokens, but our partners aren't supporting custom tokens anyhow. Going forward, we need to stop using `metaTokens` and use the core's new token structures instead. For now, let's just leave this as-is.